pos_deposit/static/src/js/models.js

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2024 Hunki Enterprises BV
+ * License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ */
+odoo.define("pos_deposit.models", function (require) {
+    "use strict";
+
+    const {Order, Orderline} = require("point_of_sale.models");
+    const Registries = require("point_of_sale.Registries");
+    const {Gui} = require("point_of_sale.Gui");
+    const core = require("web.core");
+    const _t = core._t;
+
+    const PosDepositOrderExtension = (Order) =>
+        class PosDepositOrder extends Order {
+            add_product(product, options) {
+                if (product.deposit_product_id) {
+                    const deposit =
+                        this.pos.db.product_by_id[product.deposit_product_id[0]];
+                    if (!deposit) {
+                        Gui.showPopup("ErrorPopup", {
+                            title: _t("Deposit not available"),
+                            body: _t(
+                                "The product is configured as having a deposit but the deposit product is not available in POS."
```

Comment:
UX nitpicking : I guess ``product.deposit_product_id[1]`` contains the name of the deposit. we could add it in the message, so the user know directly which product he has to enable for the PoS.